README.md

```diff
@@ -6,11 +6,22 @@ A curated list of awesome machine learning frameworks and algorithms that work o
 
 <!-- MarkdownTOC depth=4 -->
 - [Digests](#digests)
-- [Articles](#articles)
-    - [Papers](#papers)
-    - [Posts](#posts)
-    - [Talks](#talks)
-- [Software](#software)
+- [Papers](#papers)
+    - [Program Synthesis and Induction](#program-synthesis-induction)
+    - [Source Code Analysis and Language modeling](#code-analysis)
+    - [Neural Network Architectures and Algorithms](#rnn-algo)
+    - [Program Translation](#program-translation)
+    - [Code Suggestion and Completion](#code-suggestion-completion)
+    - [Program Repair and Bug Detection](#program-repair-bug-detection)
+    - [APIs and Code Mining](#api-mining)
+    - [Code Optimization](#code-optimization)
+    - [Topic Modeling](#topic-modeling)
+    - [Code Summarization](#code-summarization)
+    - [Clone Detection](#clone-detection)
+    - [Differentiable Interpreters](#differentiable-interpreters)
+- [Posts](#posts)
+- [Talks](#talks)
+- [Softwares](#softwares)
```

Comment:
"Softwares" is impossible, it is always singular. Unless you write "warez" or similar